library/ssl_srv.c

```diff
@@ -3790,15 +3790,19 @@ static int ssl_parse_certificate_verify( mbedtls_ssl_context *ssl )
     /* Read the message without adding it to the checksum */
```

Comment:
Pre-existing, only for reference for future cleanups: I think we have a layering violation here, and we should re-arrange code to only ever call `mbedtls_ssl_read_record()` and never the lower-level `mbedtls_ssl_read_record_layer()+handle_message_type()` (which ideally would be made static when we get the chance to do so). I think this could be achieved simply by calling `ssl->handshake->calc_verify()` before we read the message, but this should be double-checked as maybe it relies on state that has been modified in the meantime.

Reply:
@mpg I intend to unify this in the forthcoming reworkings by setting a flag for each message indicating whether it contributes to the checksum or not, and to do the checksum update at the _end_ of message processing by the logic layer.

Reply:
Sounds good!